### PR TITLE
Add --load-restrictor flag to Kustomize build

### PR DESCRIPTION
**Pull request by Camila-B**
* Adds `--load-restrictor=LoadRestrictionsNone` to the Kustomize build to allow Kustomizations to read files outside of the root directory
* Adds e2e test coverage for the change by modifying existing e2e hydration test data
* b/342237927

### CODE REVIEW

**Comment by Camila-B** [comment]
@tiffanny29631  Sure, done

File: scripts/install-kustomize.sh [hunk 1]
```diff
@@ -62,6 +62,7 @@ echo "$(cat "${KUSTOMIZE_CHECKSUM}")  ${KUSTOMIZE_TARBALL}" | sha256sum -c
 echo "Installing kustomize"
 mkdir -p "${STAGING_DIR}"
 tar -zxvf "${KUSTOMIZE_TARBALL}" -C "${STAGING_DIR}"
+chmod a+x "${STAGING_DIR}/kustomize"
```
Comment (sdowell) [inline scripts/install-kustomize.sh:65]: This can be submitted as a separate change

Comment (Camila-B) [reply scripts/install-kustomize.sh]: I removed the change and will create a separate PR for it

**Review by tiffanny29631 — COMMENTED** [review]
/lgtm 
I assume the mounting of empty service account was done.
See if Sam has more comments.

**Comment by Camila-B** [comment]
Yup! It was done in https://github.com/GoogleContainerTools/kpt-config-sync/pull/1243

**Review by tiffanny29631 — APPROVED** [review]

**Pull request merged**